### PR TITLE
[kmac] Add KeyMgr sideload/ data interfaces

### PR DESCRIPTION
**Pull request by eunchan**
KMAC has a feature that KeyMgr can initiates the KDF operation. To achieve this, KeyMgr and KMAC have defined the `kmac_data_req/rsp_t` interface (`typedef struct packed`) as an inter module signal form.

This PR is to add the KeyMgr interfaces including the sideloaded secret keys in KMAC/SHA3 HWIP.

This PR has the content of PR #3676 . Please take a look at the last commit only.

### CODE REVIEW

File: hw/ip/kmac/rtl/kmac.sv [hunk 1]
```diff
@@ -25,9 +25,12 @@ module kmac
...
   output tlul_pkg::tl_d2h_t tl_o,
 
-  // TODO: KeyMgr sideload (secret key) interface
+  // KeyMgr sideload (secret key) interface
+  input keymgr_pkg::hw_key_req_t keymgr_key_i,
 
-  // TODO: KeyMgr KDF data path
+  // KeyMgr KDF data path
+  input  keymgr_pkg::kmac_data_req_t keymgr_kdf_i,
+  output keymgr_pkg::kmac_data_rsp_t keymgr_kdf_o,
```
Comment (udinator) [inline hw/ip/kmac/rtl/kmac.sv:33]: i think i have two questions about this interface:

- just confirming that this is a ready/valid interface, right (or is similar)? if so, when does `kmac_data_rsp_t::done` get asserted?

- do you think there's any way to get this to work with the `hw/dv/sv/push_pull_agent`? That was developed to ideally handle all ready/valid and req/ack interfaces in the designs, but this keymgr interface seems like there is data transmission on both request and response. just from the DV perspective it might be good to be able to reuse this agent instead of having to develop something that is only slightly different.

**Comment by eunchan** [comment]
> i think i have two questions about this interface:
> just confirming that this is a ready/valid interface, right (or is similar)? if so, when does kmac_data_rsp_t::done get asserted?
> 
`done` is to indicating the digest_share0, 1 are valid. It means all KMAC operations are done and ready to use the digest. Basically it is same as absorbed indicator for SW.
> do you think there's any way to get this to work with the hw/dv/sv/push_pull_agent? That was developed to ideally handle all ready/valid and req/ack interfaces in the designs, but this keymgr interface seems like there is data transmission on both request and response. just from the DV perspective it might be good to be able to reuse this agent instead of having to develop something that is only slightly different.
> 
Even the req/rsp looks like handshaking the data each side, in reality it is not.

`valid` with `data` and `strb` are the request of the data channel. `ready` is the only response signal. `done` with `digest` are the other data path apart from the data mentioned above. Technically, they are better to be separate channels but digest cannot be used standalone and data channel is meaningless without the digest return. So they are gathered into a req/rsp channel.

TL;DR, I don't think we can re-use push_pull agent for this data path. It is more like BUS in my opinion. And still the `last` beat indicator is missing in the interface. It is likely to be revised and fine-tuned later.

**Review by udinator — APPROVED** [review]

File: hw/ip/keymgr/rtl/keymgr_pkg.sv [hunk 2]
```diff
@@ -113,20 +113,34 @@ package keymgr_pkg;
     logic valid;
     logic [KmacDataIfWidth-1:0] data;
     logic [KmacDataIfWidth/8-1:0] strb;
+    // last indicates the last beat of the data. strb can be partial only with
+    // last.
+    logic last;
```
Comment (eunchan) [inline hw/ip/keymgr/rtl/keymgr_pkg.sv:118]: @tjaychen Could you take a look at here? I added `last` in the req channel and `error` in the response channel.

Comment (tjaychen) [reply hw/ip/keymgr/rtl/keymgr_pkg.sv:118]: this looks good to me. Thanks @eunchan 
i'll make sure to hook them up internally
